src/cborpretty.c

```diff
@@ -526,7 +526,10 @@ static CborError value_to_pretty(CborStreamFunction stream, void *out, CborValue
             err = stream(out, "%s%" PRIu64 ".%s", val < 0 ? "-" : "", ival, suffix);
         } else {
             /* this number is definitely not a 64-bit integer */
-            err = stream(out, "%." DBL_DECIMAL_DIG_STR "g%s", val, suffix);
+            if (fpclassify(val) == FP_NAN)
```

Comment:
Please explain why this is needed.

Reply:
As you can see in [AppVeyor build](https://ci.appveyor.com/project/phirsov/tinycbor/builds/20867206/job/eladqc8ebdn56gnc) Visual Studio pretty-printing of NaN is somewhat weird

Reply:
I see it now, but it was working  before. What changed?

In any case, please err on the side of writing more in the commit message.

Reply:
QString::vasprintf was replaced with pure-C vsnprintf in  tests/parser/tst_parser.cpp, because this member function was introduced in the newer Qt version. vsnprintf NaN pretty-printing is broken in Visual Studio.